dvc/command/version.py

```diff
@@ -49,23 +50,24 @@ def run(self):
             # can't auto-create it, as it might cause issues if the user
             # later decides to enable shared cache mode with
             # `dvc config cache.shared group`.
-            if os.path.exists(repo.cache.local.cache_dir):
-                info.append(
-                    "Cache types: {}".format(
-                        self.get_linktype_support_info(repo)
-                    )
-                )
-                if psutil:
-                    fs_type = self.get_fs_type(repo.cache.local.cache_dir)
-                    info.append(f"Cache directory: {fs_type}")
-            else:
-                logger.warning(
-                    "Unable to detect supported link types, as cache "
-                    "directory '{}' doesn't exist. It is usually auto-created "
-                    "by commands such as `dvc add/fetch/pull/run/import`, "
-                    "but you could create it manually to enable this "
-                    "check.".format(relpath(repo.cache.local.cache_dir))
-                )
+            # Create temporary cache to determine cache type and delete
+            # it immediately when information retrieved.
+            create_tmp_cache_dir = not os.path.exists(
+                repo.cache.local.cache_dir
+            )
+            if create_tmp_cache_dir:
+                os.makedirs(repo.cache.local.cache_dir)
+                tempfile.mkstemp(dir=repo.cache.local.cache_dir)
+
+            info.append(
+                "Cache types: {}".format(self.get_linktype_support_info(repo))
+            )
+            if psutil:
+                fs_type = self.get_fs_type(repo.cache.local.cache_dir)
+                info.append(f"Cache directory: {fs_type}")
+
+            if create_tmp_cache_dir:
+                shutil.rmtree(repo.cache.local.cache_dir, ignore_errors=True)
```

Comment:
A bit worried about this racing in shared cache scenarios, where cache dir might be accidentally deleted. Also, we might simply not have any rights to create that cache_dir, so btter to wrap `makedirs` above in a try&except logic. Also, we will be leaving traces in a form of parent directories that didn't exist before. But if we are not able to create even a temporary dir, then we will still need to show some message about it, which is actually the same thing as was suggested in the original ticket by @shcheklein as a first option. That makes me wonder if we really shouldn't overcomplicate this and just stick to the first option and just not even try to create a temporary directory.

Reply:
Thanks for detailed explanation about shared cache scenario.
Just to be sure I understood your message correctly. The final solution is to remove warning completely. No new code required. Is it correct?

Reply:
@IvanRubanov Yes, I think removing it and, for example, adding
```
Cache types: https://error.dvc.org/unable-to-detect-cache-type
```
to the `dvc version`  output would be the best overall solution here in terms of complexity and effectiveness. Thank you so much for looking into this!

Reply:
Done.